### PR TITLE
PR: Make several varexp editors stay on top

### LINKED ISSUE #9051: Variable explorer viewers go to the background on macOS
### Issue Report Checklist

* [ ] Searched the [issues page](https://github.com/spyder-ide/spyder/issues?q=is%3Aissue) for similar reports
* [ ] Read the relevant sections of the [Spyder Troubleshooting Guide](https://github.com/spyder-ide/spyder/wiki/Troubleshooting-Guide-and-FAQ) and followed its advice
* [ ] Reproduced the issue after updating with ``conda update spyder`` (or ``pip``, if not using Anaconda)
* [ ] Could not reproduce inside ``jupyter qtconsole`` (if console-related)
* [ ] Tried basic troubleshooting (if a bug/error)
    * [ ] Restarted Spyder
    * [ ] Reset preferences with ``spyder --reset``
    * [ ] Reinstalled the latest version of [Anaconda](https://www.anaconda.com/download/)
    * [ ] Tried the other applicable steps from the Troubleshooting Guide
* [ ] Completed the **Problem Description**, **Steps to Reproduce** and **Version** sections below

## Problem Description
Hi there, using spyder on both windows and mac os, I noticed an odd behaviour compared on Mac OS comapred to the windows version.

On windows, I can easily open and explore different variables at the same time for comparison purpose. On Mac OS, when i open a second variable window, the first one would disapear behind spyder windows.

### What steps reproduce the problem?

1. Create at least two array, dataframe,list....
2. Inspect them with the variables explorer on Mac OS
3. 

### Paste Traceback/Error Below (if applicable)

```python-traceback

PASTE TRACEBACK HERE

```

## Versions

* Spyder version: last one
* Python version: 
* Qt version: 
* PyQt version: 
* Operating System name/version: 

### Dependencies

```
PASTE DEPENDENCIES HERE
```

### PR DESCRIPTION
**Pull request by goanpeca**
## Description of Changes

* [x] Included a screenshot (if affecting the UI)

On Mac we need to use `Qt.Tool` to obtain the desired behavior.

varexp

### Issue(s) Resolved

Fixes #9051

### Affirmation

By submitting this Pull Request or typing my (user)name below,
I affirm the [Developer Certificate of Origin](https://developercertificate.org)
with respect to all commits and content included in this PR,
and understand I am releasing the same under Spyder's MIT (Expat) license.

I certify the above statement is true and correct: @goanpeca

### CODE REVIEW

File: spyder/plugins/variableexplorer/widgets/collectionseditor.py [hunk 1]
```diff
@@ -66,6 +66,8 @@
             DataFrameEditor)
 
 
+MAC = sys.platform == 'darwin'
```
Comment (ccordoba12) [inline spyder/plugins/variableexplorer/widgets/collectionseditor.py]: Please remove this constant. We use `sys.platform == 'darwin'` wherever is needed instead.

**Comment by ccordoba12** [comment]
@goanpeca, let's move this to the 3.x branch (as you suggested) because it's a very simple fix.

**Comment by goanpeca** [comment]
Done @ccordoba12 !

**Comment by ccordoba12** [comment]
Thanks! I'll merge when everything is green.

**Comment by goanpeca** [comment]
Everything was green :-p

**Comment by goanpeca** [comment]
@ccordoba12 question: For bubbling this up to master is this what we do:

```
git checkout master
git merge 3.x
```

? or how do we move this specifc patch up to master?

**Comment by ccordoba12** [comment]
>  For bubbling this up to master is this what we do:
git checkout master
git merge 3.x

Yep, that's right, and finally `git push upstream master`. I also solve the conflicts locally before pushing and run `python runtests.py` in case there are new tests (although that's not the case here ;-).

Please do that too and use the following message:

> Merge from 3.x: PR #9119
>
> Fixes #9051

That's also necessary to make Github close the associated issue (GH only close issues when you push to master).

**Comment by goanpeca** [comment]
Got it

**Comment by ccordoba12** [comment]
The idea is to do it after each PR. I already did it, you can do it next time.

**Pull request merged**